### PR TITLE
Multiple updates + minimal test suite

### PR DESCRIPTION
**Pull request by paluh**

### CODE REVIEW

**Comment by joneshf** [comment]
To be perfectly honest with you, I don't remember creating this repo :sweat_smile:. Thanks for bringing it up to date, though.

I'll review this when I get a chance, but it'll take me a minute to get up to speed with what the code did before and to review the changes.

**Comment by paluh** [comment]
Hi,

This lib is used by `purescript-in-purescript`. I'm during the process of extracting [`JS.AST`](https://github.com/paluh/purescript-js-ast) along with its pretty printer from there. It uses this interesting `Pattern` for output code formatting :-)
P.S.
I've added some tests cases and pointed from README to @paf31 post on the topic ;-)

File: bower.json [hunk 1]
```diff
@@ -18,12 +18,7 @@
     "js"
   ],
   "dependencies": {
-    "purescript-control": "*",
```
Comment (joneshf) [inline bower.json:21]: I'd like to keep the explicit dependencies. I'm more than happy to accept this PR and make the changes afterward. So, there's nothing you need to do in this PR. If you feel up to adding the dependencies that's fine as well. I just wanted to let you know up front that they will be added back, so it didn't seem passive aggressive.

File: package.json [hunk 2]
```diff
@@ -5,5 +5,8 @@
     "grunt": "~0.4.4",
     "grunt-purescript": "~0.4.0",
     "grunt-contrib-clean": "~0.5.0"
+  },
+  "dependencies": {
+    "pulp": "^12.4.2"
```
Comment (joneshf) [inline package.json:8]: I'm probably going to move this to `devDependencies`. Similar to the `bower.json` comment, you don't have to do anything about it if you don't want to. Mentioning here so you're aware beforehand.

File: src/Text/Pretty/PatternArrows.purs [hunk 3]
```diff
@@ -111,14 +110,11 @@ data Operator u a r = Operator (forall op. (Op op) => op u a r)
 -- |
 -- A table of operators
 --
-data OperatorTable u a r = OperatorTable [[Operator u a r]]
+data OperatorTable u a r = OperatorTable (Array (NonEmpty Array (Operator u a r)))
```
Comment (joneshf) [inline src/Text/Pretty/PatternArrows.purs:115]: :heart_eyes_cat: This is much clearer. Thanks!

File: test/Main.purs [hunk 4]
```diff
@@ -0,0 +1,69 @@
+module Test.Main where
```
Comment (joneshf) [inline test/Main.purs:1]: Thanks for adding this! Can we delete `Examples.purs.hs` now? It seems like this is an equivalent file. You don't have to remove it if you'd prefer not to. But you can if you want.

**Review by joneshf — APPROVED** [review]
This looks great! Thanks for bringing things up to date.

I left a couple comments, None of them are blocking, but I'll leave this open until you see the review.

**Comment by paluh** [comment]
@joneshf Thanks for your kind review :-)

I've cleaned up dependencies as you suggested.

Provided tests are really translation of the existing example so probably we can drop it.

**Comment by paluh** [comment]
Oh, sorry I've probably misread your commend regarding explicit dependencies. Could you explain a bit more your dependency management policy? Would you like to list all transitive dependencies of prelude?

**Comment by joneshf** [comment]
No worries. Yeah, I want to try and list all the dependencies we use explicitly even if they will get picked up by globs. E.g. we use some stuff from transformers, so I'd like to explicitly list that as a dependency. I'll handle all that in future commits.

Thanks for the update!

**Comment by joneshf** [comment]
I'll make a release either later today or tomorrow morning.

**Comment by joneshf** [comment]
Actually, I don't know if I can do that.

I think the reason I don't remember this code is because I didn't write it. I'm pretty sure this repo was a fork that I never deleted. The source repo was in the `purescript` organization, and I'd guess it has dibs on the bower name. I think if you want this published on bower, you'll have to do a bit of work to figure out whether it can be registered/updated and probably maintain it going forward.

**Comment by paluh** [comment]
> I think if you want this published on bower, you'll have to do a bit of work to figure out whether it can be registered/updated and probably maintain it going forward.

Maintenance won't be a problem for me but I'm not able to find any other "current" maintainer of this package than you :-P You have the last repository of this project on github and it seems that it is missing also from bower registry.
`LICENSE` file contains only `Copyright (c) 2014 PureScript` but from commits history I see that you, @garyb and @paf31 are the authors.

Could you suggest me anyone who should I ask for publish and maintenance permission?

**Comment by garyb** [comment]
I've `bower unregister`ed the original `purescript-pattern-arrows`, so you can do so now 🙂

**Comment by paluh** [comment]
Thanks @garyb!

@joneshf so I'm going to publish and maintain this lib. Is it ok for you?

**Comment by joneshf** [comment]
Yep! Go for it!

**Comment by joneshf** [comment]
@paluh did you figure things out? I don't see anything when searching with bower, but I might be looking in the wrong place. I'd like to delete this fork, but don't want to leave you hanging in case there's still stuff to do.

**Comment by paluh** [comment]
@joneshf Sorry for the late response (was on the three weeks computer detox ;-))!
I'm ashamed to tell that I've not published my fork yet. Please remove your fork now if you want to as I don't plan to drop this maintenance of this lib and promise to push it to bower/pursuit this week.

**Comment by paluh** [comment]
@joneshf 
Sorry for this huge delay. I've just published this package and I'm going to maintain it:
https://pursuit.purescript.org/packages/purescript-pattern-arrows/0.1.0

**Pull request merged**